services/httpd/handler_test.go

```diff
@@ -318,6 +318,49 @@ func TestHandler_Ping(t *testing.T) {
 	}
 }
 
+// Ensure the handler returns the version correctly from the different endpoints.
+func TestHandler_Version(t *testing.T) {
+	h := NewHandler(false)
+	w := httptest.NewRecorder()
+	tests := []struct {
+		method   string
+		endpoint string
+		body     *bytes.Reader
+	}{
+		{
+			method:   "GET",
+			endpoint: "/ping",
+			body:     nil,
+		},
+		{
+			method:   "GET",
+			endpoint: "/query?db=foo&q=SELECT+*+FROM+bar",
+			body:     nil,
+		},
+		{
+			method:   "POST",
+			endpoint: "/write",
+			body:     bytes.NewReader(make([]byte, 10)),
+		},
+	}
+
+	for _, test := range tests {
+		if test.body != nil {
```

Comment:
@rossmcdonald Oh I see what's going on here now..

So in your example setup you should make `body` an `io.Reader`, rather than a `*bytes.Buffer`. You can obviously still use exactly the same test examples because  your `*bytes.Buffer` on 343 satisfies the `io.Reader` interface.